dll/win32/shell32/utils.cpp

```diff
@@ -762,6 +762,133 @@ SHCreatePropertyBag(_In_ REFIID riid, _Out_ void **ppvObj)
     return SHCreatePropertyBagOnMemory(STGM_READWRITE, riid, ppvObj);
 }
 
+// The helper function for SHGetUnreadMailCountW
+static LSTATUS
+SHELL_ReadSingleUnreadMailCount(
+    _In_ HKEY hKey,
+    _Out_opt_ PDWORD pdwCount,
+    _Out_opt_ PFILETIME pFileTime,
+    _Out_writes_opt_(cchShellExecuteCommand) LPWSTR pszShellExecuteCommand,
+    _In_opt_ INT cchShellExecuteCommand)
+{
+    DWORD dwType, dwCount, cbCount = sizeof(dwCount);
+    LSTATUS error = RegQueryValueExW(hKey, L"MessageCount", 0, &dwType, (PBYTE)&dwCount, &cbCount);
```

Comment:
Not using SHQueryValueExW here and below as well?

Reply:
Used `SHQueryValueExW` as well.